collect_app/src/main/java/org/odk/collect/android/widgets/items/SelectOneFromMapDialogFragment.kt

```diff
@@ -33,73 +23,40 @@ import org.odk.collect.geo.selection.MappableSelectItem
 import org.odk.collect.geo.selection.SelectionMapData
 import org.odk.collect.geo.selection.SelectionMapFragment
 import org.odk.collect.geo.selection.SelectionMapFragment.Companion.REQUEST_SELECT_ITEM
-import org.odk.collect.material.MaterialFullScreenDialogFragment
 import javax.inject.Inject
 
-class SelectOneFromMapDialogFragment(private val viewModelFactory: ViewModelProvider.Factory) :
-    MaterialFullScreenDialogFragment(), FragmentResultListener {
+class SelectOneFromMapDialogFragment(viewModelFactory: ViewModelProvider.Factory) :
+    WidgetAnswerDialogFragment<SelectionMapFragment>(
+        SelectionMapFragment::class,
+        viewModelFactory
+    ) {
 
     @Inject
     lateinit var scheduler: Scheduler
 
-    private val formEntryViewModel: FormEntryViewModel by activityViewModels { viewModelFactory }
-
     override fun onAttach(context: Context) {
         super.onAttach(context)
         DaggerUtils.getComponent(context).inject(this)
-
-        val formIndex = requireArguments().getSerializable(ARG_FORM_INDEX) as FormIndex
-        val selectedIndex = requireArguments().getSerializable(ARG_SELECTED_INDEX) as Int?
-        val prompt = formEntryViewModel.getQuestionPrompt(formIndex)
-        val selectionMapData = SelectChoicesMapData(resources, scheduler, prompt, selectedIndex)
-
-        childFragmentManager.fragmentFactory = FragmentFactoryBuilder()
-            .forClass(SelectionMapFragment::class.java) {
-                SelectionMapFragment(
-                    selectionMapData,
-                    skipSummary = Appearances.hasAppearance(prompt, Appearances.QUICK),
-                    zoomToFitItems = false,
-                    showNewItemButton = false,
-                    onBackPressedDispatcher = { (requireDialog() as ComponentDialog).onBackPressedDispatcher }
-                )
-            }
-            .build()
-
-        childFragmentManager.setFragmentResultListener(REQUEST_SELECT_ITEM, this, this)
-    }
-
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View {
-        val binding = SelectOneFromMapDialogLayoutBinding.inflate(inflater)
-        return binding.root
     }
 
-    override fun getToolbar(): Toolbar? {
-        return null
-    }
-
-    override fun onBackPressed() {
-        dismiss()
-    }
-
-    override fun onCloseClicked() {
-        // No toolbar so not relevant
-    }
+    override fun onCreateFragment(prompt: FormEntryPrompt): SelectionMapFragment {
+        childFragmentManager.setFragmentResultListener(REQUEST_SELECT_ITEM, this) { _, result ->
+            val selectedIndex = result.getLong(SelectionMapFragment.RESULT_SELECTED_ITEM).toInt()
```

Comment:
It could be getInt so that we don't have to convert to int here?

Reply:
It looks like `getInt` doesn't handle the type conversion properly, and we always end up with `0`. If you change it, you'll see that a test in `SelectOneFromMapDialogFragmentTest` fails.